tests/unit/controllers/problem.test.ts

```diff
@@ -1,40 +1,31 @@
 import sinon from "sinon";
-import { IProblem, Difficulty, PlatformName } from "../../../src/interfaces/IProblem";
-import { IProblemSet } from "../../../src/interfaces/IProblemSet";
 import { problemController } from "../../../src/controllers/problem";
 import { problemDBInteractionsStubs } from "../stubs/problem";
 import { problemSetDBInteractionsStubs } from "../stubs/problemSet";
 import { mockReq, mockRes } from "sinon-express-mock";
-import chai, { expect } from "chai";
-
-let problemDBStubs;
-let problemSetDBStubs;
-
-const testProblemSet: IProblemSet = {
-    title: "Test Problem Set",
-    description: "Test problem set description.",
-    tags: ["Dynamic programming", "Test"],
-    problemCount: 1
-};
-const testProblem: IProblem = {
-    title: "Test Problem",
-    problemId: "27796f7b6717753aeee14bb4ba2acf6f55c88956",
-    source: PlatformName[0],
-    sourceLink: "https://codeforces.com/problemset/problem/1/A",
-    problemSetIds: [],
-    problemMetadata: {
-        platformProblemId: "1A",
-        difficulty: Difficulty[0]
-    }
+import { validatorStubs, validationErrorWithMessage, emptyValidationError } from "../stubs/misc"
+import { testProblemModel1 } from "../../util/sampleData";
+
+// Initialized outside of "describe" blocks to ensure typesafety + intellisense
+let stubs = {
+    problemDB: problemDBInteractionsStubs(),
+    problemSetDB: problemSetDBInteractionsStubs(),
+    validators: validatorStubs()
 };
+stubs.problemDB.restore()
+stubs.problemSetDB.restore()
+stubs.validators.restore()
```

Comment:
hmm do we really need that? intellisense would still work with old code. I agree these lines ensure typesafety, but that would only be useful if stubs are being reassigned inside the tests, but we are just accessing them inside the tests, so idk. To me it looks like excessive code with no real use case(object reassignment), but maybe @MohamedBeydoun can give his opinion on that as well.

Reply:
I think its beneficial  to keep a contract at the top of the tests and clearly state the dependencies. Its not overly verbose either imo